tasks/create_python_venv.yml

```diff
@@ -8,14 +8,36 @@
           - virtualenv
       when: not assessment_tool_python2
 
-    - name: Install virtualenv via pip
-      ansible.builtin.pip:
-        name:
-          # virtualenv 20.22.0 removed support for creating Python 2
-          # venvs:
-          # https://virtualenv.pypa.io/en/latest/changelog.html#v20-22-0-2023-04-19
-          - virtualenv<20.22.0
+    # The shadiness is necessary because, on modern distributions, pip
+    # (correctly) refuses to uninstall Python packages installed via
+    # system packages.
+    - name: Shadily install virtualenv for Python 2 via pip
       when: assessment_tool_python2
+      tags:
+        # This kind of chicanery can never pass idempotence.
+        - molecule-idempotence-notest
+      block:
+        # We can revert to the system packages that provide virtualenv
+        # and platformdirs once the Python 2 virtual environment is
+        # created.
+        - name: >-
+            Shadily uninstall some Python 3 packages via the package
+            manager
+          ansible.builtin.package:
+            name:
+              - python3-platformdirs
+              - virtualenv
+            state: absent
+          register: shady_uninstall
+        - name: Shadily install necessary packages for Python 2 via pip
+          ansible.builtin.pip:
+            name:
+              - platformdirs
+              # virtualenv 20.22.0 removed support for creating Python 2
+              # venvs:
+              # https://virtualenv.pypa.io/en/latest/changelog.html#v20-22-0-2023-04-19
+              - virtualenv<20.22.0
+          register: shady_install
 
 - name: Create the virtualenv
```

Comment:
Just as a thought would it be more straightforward to install `virtualenv` into the Python 2 environment and use the Python 2 `pip` path when creating the virtual environment?

Reply:
I don't think I'm following you.  `ansible.builtin.pip` uses the `virtualenv` package to create the Python virtual environment.

Reply:
`python-virtualenv` only exists on Buster; all other supported releases only offer `python3-virtualenv`.

Reply:
Sorry I meant installing `virtualenv` into the Python 2 environment and using that environment for setup. However, I see that getting `pip` installed in Python 2 for Kali Linux is problematic. It looks like it would be even more work to try and bootstrap the Python 2 environment to spin up its own virtual environments.

> [!WARNING]
> When Kali Linux changes to follow [PEP 668](https://peps.python.org/pep-0668/), [which is on the horizon](https://www.kali.org/blog/python-externally-managed/), the workaround in this PR will not work.

Reply:
Given that there is no clear path forward right now, are you OK if I just create an issue from your comment for now?

Reply:
See issue #56.

Reply:
Works for me.